pkg/controller/jobframework/reconciler.go

```diff
@@ -505,6 +515,21 @@ func (r *JobReconciler) equivalentToWorkload(job GenericJob, object client.Objec
 	return true
 }
 
+func (r *JobReconciler) updateWorkloadToMatchJob(ctx context.Context, job GenericJob, object client.Object, wl *kueue.Workload) (*kueue.Workload, error) {
+	newWl, err := r.constructWorkload(ctx, job, object)
+	if err != nil {
+		return nil, fmt.Errorf("can't construct workload for update: %w", err)
+	}
+	wl.Spec = newWl.Spec
+	if err = r.client.Update(ctx, wl); err != nil {
```

Comment:
So the key point here is we'll replace the unadmitted workload with the newest spec. right?

Reply:
Correct. However note, that currently we can update only the amount of pods/containers: https://github.com/kubernetes-sigs/kueue/issues/1264